tests/spec/s-function.js

```diff
@@ -1,7 +1,177 @@
 /* global describe, it, expect, beforeEach */
+var hasStrictMode = (function () {
+  'use strict';
+
+   return !this;
+}());
+
+var global = (function () {
+   return this;
```

Comment:
I'm thinking more that this needs to be a test - since detecting the global object is very difficult.

Alternatively, you could use `var global = Function('return this')();` which is unassailably correct everywhere.